src/main/java/io/myzticbean/finditemaddon/Utils/WarpUtils/PlayerWarpsUtil.java

```diff
@@ -22,54 +22,49 @@
 import io.myzticbean.finditemaddon.Dependencies.PlayerWarpsPlugin;
 import io.myzticbean.finditemaddon.FindItemAddOn;
 import io.myzticbean.finditemaddon.Utils.CommonUtils;
-import io.myzticbean.finditemaddon.Utils.LoggerUtils;
 import org.bukkit.Location;
 import org.jetbrains.annotations.Nullable;
 
-import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
 import java.util.TreeMap;
+import java.util.UUID;
 
 /**
- * @author myzticbean
+ * @author myzticbean & lukemango
  */
 public class PlayerWarpsUtil {
 
     @Nullable
-    public Warp findNearestWarp(Location shopLocation) {
-        List<Warp> allWarps = PlayerWarpsPlugin.getAllWarps();
-        if(!allWarps.isEmpty()) {
+    public Warp findNearestWarp(Location shopLocation, UUID shopOwner) {
+        List<Warp> playersWarps = PlayerWarpsPlugin.getAllWarps().stream()
+                .filter(warp -> warp.getWarpLocation().getWorld() != null)
+                .filter(warp -> warp.getWarpLocation().getWorld().equals(shopLocation.getWorld().getName()))
+                .filter(warp -> warp.getWarpPlayer().getUUID().equals(shopOwner))
```

Comment:
@lukemango This will filter the player warps for the player who is searching. I am thinking, what if someone doesnt want to restrict this, and wants it to look through all public warps

Reply:
So, you could make it as a boolean flag in config.

Reply:
Let me merge this, and I will add those changes in my repo.